### ISSUE TITLE
Game going non-responsive after Cut Tree Trunk into Planks action is performed.

### ISSUE THREAD

**Issue by Shadw21** [CleverRaven/Cataclysm-DDA#32587]
Game version: 0.D-5823-g763c25c (tiles)

Operating system: Win 10 Pro Version 10.0.17763

Tiles or curses: tiles

Mods active:
Dark Days Ahead
Disable NPC Needs
Makeshift Items Mod
More Survival Tools
Modular Turrets
Salvaged Robots
Hydroponics
Mining Mod
Jury-Rigged Robots
StatsThoughSkills IV

### Expected behavior
Generate planks and other wood items from a tree trunk.
Game continues to run after action completes, allowing you to repeat the action or do something else in game.

### Actual behavior
After planks and other wood items generate, with in game time passing in 5 minute increments, and the message screen showing what was created, the in game time continues to progress as if the Period key "." was being held down. 
After 2-10 seconds game goes non-responsive and must be force closed.

### Steps to reproduce the behavior
Press Shift-8 or asterisk "*" to open Construction Menu.
Find Cut Tree Trunk into Planks in the menu with a with Stone Adze or Stone Hand Axe wielded or in your inventory.
Select direction of the tree trunk or tree stump to start the action.
Wait for action to complete, wood products generate in the message log, and the in game time to pass until game crashes.

### Additional notes
Tested multiple times on two computers across a few different builds, 9363, 9365, and 9366 that released on 7/23. The latest test being on Build 9366 with a fresh world and character, world settings and mods copied from a pre-existing world. 

I believe this was working for me on one of the builds from 7/21 or earlier.

**Comment by roaringjohn** [comment]
You do know StatsThroughSkills IV is lua, and that lua has been discontinued, right?
Probably unrelated to your problem, but still.

**Comment by Contemplator11** [comment]
> You do know StatsThroughSkills IV is lua, and that lua has been discontinued, right?
> Dunno if it's related to your problem, but still.

I had the same issue, playing without STS.

**Comment by Shadw21** [comment]
>You do know StatsThroughSkills IV is lua, and that lua has been discontinued, right?
Dunno if it's related to your problem, but still.


Had no idea, I only learned about the game...  last June/July actually, when it was still 0.C. I'd just carried over the same mods I'd had then while trying to learn the game. Just started really getting into it again last week. I'll gen a new world without STSIV or any other mods that I added in and test it out.

**Comment by nexusmrsep** [comment]
Could not reproduce in vanilla.

**Comment by Shadw21** [comment]
I'm getting the same thing happening with only the "Dark Days Ahead" and "Disable NPC Needs" mods being active in a newly generated world.

I'm not sure if that's considered vanilla or not, but those are the default mods in a new world.

**Comment by LeoPersson** [comment]
Just tried this in a 100% vanilla, fresh, world and was able to reproduce the bug. Both the [*] construction and [a] activation methods have the same result.

If you leave the game as "not responding" for a while it continues like it's supposed to. Potential memory issue? My rig is decently powerful so that may be why it pulls through eventually.

**Comment by Shadw21** [comment]
Fresh install, new world, default everything, reproduced it.

I haven't let it run in the not responsive state for more than a minute or so I think.

Just tested again while typing this up, it will indeed resolve itself, but will take a couple of minutes to do so.

Starting a construction at 12:55:30PM in game, the construction completed at about 2:00:00PM, then it switched to ticking up rapidly, second by second, until it froze at 02:05:47PM. It has however been non responsive for even longer this time. When it finally started responding again, the in game time was 3:10:39PM.

**Comment by glenmack** [comment]
Happened in my world, though mine was progressing in single second increments. Leaving it to pull through also worked instead of it crashing for me.

**Comment by nexusmrsep** [comment]
I'll try to check it out again later. I might have used a slightly outdated version where it worked. We'll see.

**Comment by nexusmrsep** [comment]
Confirmed on recent build. While construction takes place time passes as expected, but afterwards, after the construction products are received it loops/processes "something" on 1s/tick basis that can certainly be interpret as a freeze. It's finite - ends after a while. Using stone adze, crafting part took about from 19:00 to around 20:30 (in 5 min intervals) and looped part it took from 20:30 to 21:30 (in 1s intervals).

**Comment by nexusmrsep** [comment]
Culprit: 
![image](https://user-images.githubusercontent.com/39925111/61890075-011fbd80-af07-11e9-8bb3-f033dd180913.png)

**Comment by nexusmrsep** [comment]
Culprit: #32558

**Comment by pjf** [comment]
Even with #32558 applied this still happens for me on a4acee28f41f741baca796b20fef4bb3246a0df7. The game quickly finishes making the planks, and then goes second-by-second for an hour or so before becoming responsive again.

**Comment by anothersimulacrum** [comment]
#32588 Is the one causing the problem, and it hasn't been fixed yet, so that's why.

**Comment by pjf** [comment]
@anothersimulacrum : I'm not quite sure I follow. I have #32588 applied, and still see the same behaviour after attempting to chop a tree trunk into planks.

**Comment by anothersimulacrum** [comment]
Sorry, typo, I meant #32558. That's what causing this bug. This bug has not been fixed yet, so you will be experiencing it unless you revert that.

**Comment by pjf** [comment]
D'oh! For some reason I thought #32558 *fixed* this bug, rather than resulting in it. This is what happens when I haven't my coffee in the morning. Thanks for your patience with me, @anothersimulacrum. :)

**Comment by pjf** [comment]
Reporting in that I also see this behaviour with hand-saws to turn logs into planks, as well as using the construction menu (`*`).

**Comment by Soup-de-Loop** [comment]
Just wanted to leave a note that I managed to reproduce the issue in several recent versions in experimental. I would cut planks normally with a wood saw, then alt-tab and window to try and switch to another program. Once I successfully had CDDA in the background, it would hang and windows would report such and prompt to close or wait for a response.

In-game behaviour was also interesting. At 10+ fab, I was making 10 planks out of one log, and it took exactly 10 ingame minutes if the game didn't hang. If I got it to hang, it would take longer, maybe a minute or so (different values). Other variables include: the planks are created immediately, the game then waits turn by turn (in seconds) until the crafting time is complete; I could not interrupt the action by pressing any button, escape, . etc don't interrupt- but every key I pressed while waiting was processed immediately after the waiting time is complete. Unsure if the game would interrupt it if an enemy approached during that time or if they'd get the 10 minutes of uninterrupted attacks on the character.
